source/common/router/scoped_rds.cc

```diff
@@ -453,9 +453,10 @@ ScopedRdsConfigSubscription::detectUpdateConflictAndCleanupRemoved(
       exception_msg = fmt::format("duplicate scoped route configuration '{}' found", scope_name);
       return clean_removed_resources;
     }
-    const envoy::config::route::v3::ScopedRouteConfiguration& scoped_route_config =
+    envoy::config::route::v3::ScopedRouteConfiguration scoped_route_config =
```

Comment:
For defense in depth, please verify that `scope_name_by_hash_.erase(iter->second->scopeKey().hash());` is able to find the element being erased.

Reply:
makes sense.